src/loop_libraries/loop_library_interface.hpp

```diff
@@ -0,0 +1,59 @@
+// ====================================================================
+// This file is part of FlexibleSUSY.
+//
+// FlexibleSUSY is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published
+// by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// FlexibleSUSY is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with FlexibleSUSY.  If not, see
+// <http://www.gnu.org/licenses/>.
+// ====================================================================
+
+#ifndef LOOP_LIBRARY_INTERFACE
+#define LOOP_LIBRARY_INTERFACE
+
+#include <complex>
+#include <boost/preprocessor/seq/for_each.hpp>
+
+#define ARGS_TYPE(R,COMMA,ELEM) std::complex<double> ELEM,
+#define VIRTUAL(R,ARGS,NAME) virtual std::complex<double> NAME ARGS = 0;
+
+#define A_ARGS_SEQ (m02_in)
+#define B_ARGS_SEQ (p10_in)(m02_in)(m12_in)
+#define C_ARGS_SEQ (p10_in)(p21_in)(p20_in)(m02_in)(m12_in)(m22_in)
+#define D_ARGS_SEQ (p10_in)(p21_in)(p32_in)(p30_in)(p20_in)(p31_in)(m02_in)(m12_in)(m22_in)(m32_in)
+
+#define A_ARGS BOOST_PP_SEQ_FOR_EACH(ARGS_TYPE,,A_ARGS_SEQ) double scl2_in
+#define B_ARGS BOOST_PP_SEQ_FOR_EACH(ARGS_TYPE,,B_ARGS_SEQ) double scl2_in
+#define C_ARGS BOOST_PP_SEQ_FOR_EACH(ARGS_TYPE,,C_ARGS_SEQ) double scl2_in
+#define D_ARGS BOOST_PP_SEQ_FOR_EACH(ARGS_TYPE,,D_ARGS_SEQ) double scl2_in
+
+#define A_SEQ (A0)
+#define B_SEQ (B0)(B1)(B00)
+#define C_SEQ (C0)(C1)(C2)(C00)(C11)(C12)(C22)
+#define D_SEQ (D0)(D00)(D1)(D11)(D12)(D13)(D2)(D22)(D23)(D3)(D33)
+
+namespace looplibrary
+{
+class Loop_library_interface
```

Comment:
Please add a detailed description of this interface class and its methods, including a short usage example.

The description must contain:

* The expanded macros, which expand to function definitions. This is necessary so the user knows what to call.
* The description of the arguments of all methods. In particular information about masses and momenta, squared or non-squared, the array entries etc. Probably, equations would be convenient to describe the functions and the arguments.
* A documentation of the returned loop functions, including references to papers. This is important in order to define the various sign conventions etc.
* A short example usage.

Reply:
Done in 53b3245.